File: google/cloud/osconfig/agentendpoint/v1/osconfig-agentendpoint-v1-py/google/cloud/osconfig/agentendpoint_v1/types/inventory.py
# -*- coding: utf-8 -*-
# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
import proto  # type: ignore

from google.protobuf import timestamp_pb2  # type: ignore


__protobuf__ = proto.module(
    package='google.cloud.osconfig.agentendpoint.v1',
    manifest={
        'Inventory',
    },
)


class Inventory(proto.Message):
    r"""The inventory details of a VM.
    Attributes:
        os_info (google.cloud.osconfig.agentendpoint_v1.types.Inventory.OsInfo):
            Base level operating system information for
            the VM.
        installed_packages (Sequence[google.cloud.osconfig.agentendpoint_v1.types.Inventory.SoftwarePackage]):
            A list of installed packages currently on the
            VM.
        available_packages (Sequence[google.cloud.osconfig.agentendpoint_v1.types.Inventory.SoftwarePackage]):
            A list of software updates available for the
            VM as reported by the update managers.
    """

    class OsInfo(proto.Message):
        r"""Operating system information for the VM.
        Attributes:
            hostname (str):
                The VM hostname.
            long_name (str):
                The operating system long name.
                For example 'Debian GNU/Linux 9' or 'Microsoft
                Window Server 2019 Datacenter'.
            short_name (str):
                The operating system short name.
                For example, 'windows' or 'debian'.
            version (str):
                The version of the operating system.
            architecture (str):
                The system architecture of the operating
                system.
            kernel_version (str):
                The kernel version of the operating system.
            kernel_release (str):
                The kernel release of the operating system.
            osconfig_agent_version (str):
                The current version of the OS Config agent
                running on the VM.
        """

        hostname = proto.Field(
            proto.STRING,
            number=1,
        )
        long_name = proto.Field(
            proto.STRING,
            number=2,
        )
        short_name = proto.Field(
            proto.STRING,
            number=3,
        )
        version = proto.Field(
            proto.STRING,
            number=4,
        )
        architecture = proto.Field(
            proto.STRING,
            number=5,
        )
        kernel_version = proto.Field(
            proto.STRING,
            number=6,
        )
        kernel_release = proto.Field(
            proto.STRING,
            number=7,
        )
        osconfig_agent_version = proto.Field(
            proto.STRING,
            number=8,
        )

    class SoftwarePackage(proto.Message):
        r"""Software package information of the operating system.
        Attributes:
            yum_package (google.cloud.osconfig.agentendpoint_v1.types.Inventory.VersionedPackage):
                Yum package info. For details about the yum package manager,
                see
                https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/deployment_guide/ch-yum.
            apt_package (google.cloud.osconfig.agentendpoint_v1.types.Inventory.VersionedPackage):
                Details of an APT package.
                For details about the apt package manager, see
                https://wiki.debian.org/Apt.
            zypper_package (google.cloud.osconfig.agentendpoint_v1.types.Inventory.VersionedPackage):
                Details of a Zypper package. For details about the Zypper
                package manager, see
                https://en.opensuse.org/SDB:Zypper_manual.
            googet_package (google.cloud.osconfig.agentendpoint_v1.types.Inventory.VersionedPackage):
                Details of a Googet package.
                For details about the googet package manager,
                see  https://github.com/google/googet.
            zypper_patch (google.cloud.osconfig.agentendpoint_v1.types.Inventory.ZypperPatch):
                Details of a Zypper patch. For details about the Zypper
                package manager, see
                https://en.opensuse.org/SDB:Zypper_manual.
            wua_package (google.cloud.osconfig.agentendpoint_v1.types.Inventory.WindowsUpdatePackage):
                Details of a Windows Update package. See
                https://docs.microsoft.com/en-us/windows/win32/api/_wua/ for
                information about Windows Update.
            qfe_package (google.cloud.osconfig.agentendpoint_v1.types.Inventory.WindowsQuickFixEngineeringPackage):
                Details of a Windows Quick Fix engineering
                package. See
                https://docs.microsoft.com/en-
                us/windows/win32/cimwin32prov/win32-quickfixengineering
                for info in Windows Quick Fix Engineering.
            cos_package (google.cloud.osconfig.agentendpoint_v1.types.Inventory.VersionedPackage):
                Details of a COS package.
        """

        yum_package = proto.Field(
            proto.MESSAGE,
            number=1,
            oneof='details',
            message='Inventory.VersionedPackage',
        )
        apt_package = proto.Field(
            proto.MESSAGE,
            number=2,
            oneof='details',
            message='Inventory.VersionedPackage',
        )
        zypper_package = proto.Field(
            proto.MESSAGE,
            number=3,
            oneof='details',
            message='Inventory.VersionedPackage',
        )
        googet_package = proto.Field(
            proto.MESSAGE,
            number=4,
            oneof='details',
            message='Inventory.VersionedPackage',
        )
        zypper_patch = proto.Field(
            proto.MESSAGE,
            number=5,
            oneof='details',
            message='Inventory.ZypperPatch',
        )
        wua_package = proto.Field(
            proto.MESSAGE,
            number=6,
            oneof='details',
            message='Inventory.WindowsUpdatePackage',
        )
        qfe_package = proto.Field(
            proto.MESSAGE,
            number=7,
            oneof='details',
            message='Inventory.WindowsQuickFixEngineeringPackage',
        )
        cos_package = proto.Field(
            proto.MESSAGE,
            number=8,
            oneof='details',
            message='Inventory.VersionedPackage',
        )

    class VersionedPackage(proto.Message):
        r"""Information related to the a standard versioned package.
        This includes package info for APT, Yum, Zypper, and Googet
        package managers.

        Attributes:
            package_name (str):
                The name of the package.
            architecture (str):
                The system architecture this package is
                intended for.
            version (str):
                The version of the package.
        """

        package_name = proto.Field(
            proto.STRING,
            number=1,
        )
        architecture = proto.Field(
            proto.STRING,
            number=2,
        )
        version = proto.Field(
            proto.STRING,
            number=3,
        )

    class ZypperPatch(proto.Message):
        r"""Details related to a Zypper Patch.
        Attributes:
            patch_name (str):
                The name of the patch.
            category (str):
                The category of the patch.
            severity (str):
                The severity specified for this patch
            summary (str):
                Any summary information provided about this
                patch.
        """

        patch_name = proto.Field(
            proto.STRING,
            number=1,
        )
        category = proto.Field(
            proto.STRING,
            number=2,
        )
        severity = proto.Field(
            proto.STRING,
            number=3,
        )
        summary = proto.Field(
            proto.STRING,
            number=4,
        )

    class WindowsUpdatePackage(proto.Message):
        r"""Details related to a Windows Update package. Field data and names
        are taken from Windows Update API IUpdate Interface:
        https://docs.microsoft.com/en-us/windows/win32/api/_wua/ Descriptive
        fields like title, and description are localized based on the locale
        of the VM being updated.

        Attributes:
            title (str):
                The localized title of the update package.
            description (str):
                The localized description of the update
                package.
            categories (Sequence[google.cloud.osconfig.agentendpoint_v1.types.Inventory.WindowsUpdatePackage.WindowsUpdateCategory]):
                The categories that are associated with this
                update package.
            kb_article_ids (Sequence[str]):
                A collection of Microsoft Knowledge Base
                article IDs that are associated with the update
                package.
            support_url (str):
                A hyperlink to the language-specific support
                information for the update.
            more_info_urls (Sequence[str]):
                A collection of URLs that provide more
                information about the update package.
            update_id (str):
                Gets the identifier of an update package.
                Stays the same across revisions.
            revision_number (int):
                The revision number of this update package.
            last_deployment_change_time (google.protobuf.timestamp_pb2.Timestamp):
                The last published date of the update, in
                (UTC) date and time.
        """

        class WindowsUpdateCategory(proto.Message):
            r"""Categories specified by the Windows Update.
            Attributes:
                id (str):
                    The identifier of the windows update
                    category.
                name (str):
                    The name of the windows update category.
            """

            id = proto.Field(
                proto.STRING,
                number=1,
            )
            name = proto.Field(
                proto.STRING,
                number=2,
            )

        title = proto.Field(
            proto.STRING,
            number=1,
        )
        description = proto.Field(
            proto.STRING,
            number=2,
        )
        categories = proto.RepeatedField(
            proto.MESSAGE,
            number=3,
            message='Inventory.WindowsUpdatePackage.WindowsUpdateCategory',
        )
        kb_article_ids = proto.RepeatedField(
            proto.STRING,
            number=4,
        )
        support_url = proto.Field(
            proto.STRING,
            number=5,
        )
        more_info_urls = proto.RepeatedField(
            proto.STRING,
            number=6,
        )
        update_id = proto.Field(
            proto.STRING,
            number=7,
        )
        revision_number = proto.Field(
            proto.INT32,
            number=8,
        )
        last_deployment_change_time = proto.Field(
            proto.MESSAGE,
            number=9,
            message=timestamp_pb2.Timestamp,
        )

    class WindowsQuickFixEngineeringPackage(proto.Message):
        r"""Information related to a Quick Fix Engineering package.
        Fields are taken from Windows QuickFixEngineering Interface and
        match the source names:
        https://docs.microsoft.com/en-
        us/windows/win32/cimwin32prov/win32-quickfixengineering

        Attributes:
            caption (str):
                A short textual description of the QFE
                update.
            description (str):
                A textual description of the QFE update.
            hot_fix_id (str):
                Unique identifier associated with a
                particular QFE update.
            install_time (google.protobuf.timestamp_pb2.Timestamp):
                Date that the QFE update was installed. Mapped from
                installed_on field.
        """

        caption = proto.Field(
            proto.STRING,
            number=1,
        )
        description = proto.Field(
            proto.STRING,
            number=2,
        )
        hot_fix_id = proto.Field(
            proto.STRING,
            number=3,
        )
        install_time = proto.Field(
            proto.MESSAGE,
            number=4,
            message=timestamp_pb2.Timestamp,
        )

    os_info = proto.Field(
        proto.MESSAGE,
        number=1,
        message=OsInfo,
    )
    installed_packages = proto.RepeatedField(
        proto.MESSAGE,
        number=2,
        message=SoftwarePackage,
    )
    available_packages = proto.RepeatedField(
        proto.MESSAGE,
        number=3,
        message=SoftwarePackage,
    )


__all__ = tuple(sorted(__protobuf__.manifest))

File: google/cloud/osconfig/agentendpoint/v1/google-cloud-osconfig-agentendpoint-v1-ruby/lib/google/cloud/osconfig/agentendpoint/v1/doc/google/cloud/osconfig/agentendpoint/v1/inventory.rb
# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     https://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

module Google
  module Cloud
    module Osconfig
      module Agentendpoint
        module V1
          # The inventory details of a VM.
          # @!attribute [rw] os_info
          #   @return [Google::Cloud::Osconfig::Agentendpoint::V1::Inventory::OsInfo]
          #     Base level operating system information for the VM.
          # @!attribute [rw] installed_packages
          #   @return [Array<Google::Cloud::Osconfig::Agentendpoint::V1::Inventory::SoftwarePackage>]
          #     A list of installed packages currently on the VM.
          # @!attribute [rw] available_packages
          #   @return [Array<Google::Cloud::Osconfig::Agentendpoint::V1::Inventory::SoftwarePackage>]
          #     A list of software updates available for the VM as reported by the update
          #     managers.
          class Inventory
            # Operating system information for the VM.
            # @!attribute [rw] hostname
            #   @return [String]
            #     The VM hostname.
            # @!attribute [rw] long_name
            #   @return [String]
            #     The operating system long name.
            #     For example 'Debian GNU/Linux 9' or 'Microsoft Window Server 2019
            #     Datacenter'.
            # @!attribute [rw] short_name
            #   @return [String]
            #     The operating system short name.
            #     For example, 'windows' or 'debian'.
            # @!attribute [rw] version
            #   @return [String]
            #     The version of the operating system.
            # @!attribute [rw] architecture
            #   @return [String]
            #     The system architecture of the operating system.
            # @!attribute [rw] kernel_version
            #   @return [String]
            #     The kernel version of the operating system.
            # @!attribute [rw] kernel_release
            #   @return [String]
            #     The kernel release of the operating system.
            # @!attribute [rw] osconfig_agent_version
            #   @return [String]
            #     The current version of the OS Config agent running on the VM.
            class OsInfo; end

            # Software package information of the operating system.
            # @!attribute [rw] yum_package
            #   @return [Google::Cloud::Osconfig::Agentendpoint::V1::Inventory::VersionedPackage]
            #     Yum package info.
            #     For details about the yum package manager, see
            #     https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/deployment_guide/ch-yum.
            # @!attribute [rw] apt_package
            #   @return [Google::Cloud::Osconfig::Agentendpoint::V1::Inventory::VersionedPackage]
            #     Details of an APT package.
            #     For details about the apt package manager, see
            #     https://wiki.debian.org/Apt.
            # @!attribute [rw] zypper_package
            #   @return [Google::Cloud::Osconfig::Agentendpoint::V1::Inventory::VersionedPackage]
            #     Details of a Zypper package.
            #     For details about the Zypper package manager, see
            #     https://en.opensuse.org/SDB:Zypper_manual.
            # @!attribute [rw] googet_package
            #   @return [Google::Cloud::Osconfig::Agentendpoint::V1::Inventory::VersionedPackage]
            #     Details of a Googet package.
            #      For details about the googet package manager, see
            #      https://github.com/google/googet.
            # @!attribute [rw] zypper_patch
            #   @return [Google::Cloud::Osconfig::Agentendpoint::V1::Inventory::ZypperPatch]
            #     Details of a Zypper patch.
            #     For details about the Zypper package manager, see
            #     https://en.opensuse.org/SDB:Zypper_manual.
            # @!attribute [rw] wua_package
            #   @return [Google::Cloud::Osconfig::Agentendpoint::V1::Inventory::WindowsUpdatePackage]
            #     Details of a Windows Update package.
            #     See https://docs.microsoft.com/en-us/windows/win32/api/_wua/ for
            #     information about Windows Update.
            # @!attribute [rw] qfe_package
            #   @return [Google::Cloud::Osconfig::Agentendpoint::V1::Inventory::WindowsQuickFixEngineeringPackage]
            #     Details of a Windows Quick Fix engineering package.
            #     See
            #     https://docs.microsoft.com/en-us/windows/win32/cimwin32prov/win32-quickfixengineering
            #     for info in Windows Quick Fix Engineering.
            # @!attribute [rw] cos_package
            #   @return [Google::Cloud::Osconfig::Agentendpoint::V1::Inventory::VersionedPackage]
            #     Details of a COS package.
            class SoftwarePackage; end

            # Information related to the a standard versioned package.  This includes
            # package info for APT, Yum, Zypper, and Googet package managers.
            # @!attribute [rw] package_name
            #   @return [String]
            #     The name of the package.
            # @!attribute [rw] architecture
            #   @return [String]
            #     The system architecture this package is intended for.
            # @!attribute [rw] version
            #   @return [String]
            #     The version of the package.
            class VersionedPackage; end

            # Details related to a Zypper Patch.
            # @!attribute [rw] patch_name
            #   @return [String]
            #     The name of the patch.
            # @!attribute [rw] category
            #   @return [String]
            #     The category of the patch.
            # @!attribute [rw] severity
            #   @return [String]
            #     The severity specified for this patch
            # @!attribute [rw] summary
            #   @return [String]
            #     Any summary information provided about this patch.
            class ZypperPatch; end

            # Details related to a Windows Update package.
            # Field data and names are taken from Windows Update API IUpdate Interface:
            # https://docs.microsoft.com/en-us/windows/win32/api/_wua/
            # Descriptive fields like title, and description are localized based on
            # the locale of the VM being updated.
            # @!attribute [rw] title
            #   @return [String]
            #     The localized title of the update package.
            # @!attribute [rw] description
            #   @return [String]
            #     The localized description of the update package.
            # @!attribute [rw] categories
            #   @return [Array<Google::Cloud::Osconfig::Agentendpoint::V1::Inventory::WindowsUpdatePackage::WindowsUpdateCategory>]
            #     The categories that are associated with this update package.
            # @!attribute [rw] kb_article_ids
            #   @return [Array<String>]
            #     A collection of Microsoft Knowledge Base article IDs that are associated
            #     with the update package.
            # @!attribute [rw] support_url
            #   @return [String]
            #     A hyperlink to the language-specific support information for the update.
            # @!attribute [rw] more_info_urls
            #   @return [Array<String>]
            #     A collection of URLs that provide more information about the update
            #     package.
            # @!attribute [rw] update_id
            #   @return [String]
            #     Gets the identifier of an update package.  Stays the same across
            #     revisions.
            # @!attribute [rw] revision_number
            #   @return [Integer]
            #     The revision number of this update package.
            # @!attribute [rw] last_deployment_change_time
            #   @return [Google::Protobuf::Timestamp]
            #     The last published date of the update, in (UTC) date and time.
            class WindowsUpdatePackage
              # Categories specified by the Windows Update.
              # @!attribute [rw] id
              #   @return [String]
              #     The identifier of the windows update category.
              # @!attribute [rw] name
              #   @return [String]
              #     The name of the windows update category.
              class WindowsUpdateCategory; end
            end

            # Information related to a Quick Fix Engineering package.
            # Fields are taken from Windows QuickFixEngineering Interface and match
            # the source names:
            # https://docs.microsoft.com/en-us/windows/win32/cimwin32prov/win32-quickfixengineering
            # @!attribute [rw] caption
            #   @return [String]
            #     A short textual description of the QFE update.
            # @!attribute [rw] description
            #   @return [String]
            #     A textual description of the QFE update.
            # @!attribute [rw] hot_fix_id
            #   @return [String]
            #     Unique identifier associated with a particular QFE update.
            # @!attribute [rw] install_time
            #   @return [Google::Protobuf::Timestamp]
            #     Date that the QFE update was installed.  Mapped from installed_on field.
            class WindowsQuickFixEngineeringPackage; end
          end
        end
      end
    end
  end
end